src/plugins/plugin.gridlines.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+var defaults = require('../core/core.defaults');
+var helpers = require('../helpers/index');
+var MODEL_KEY = '$gridlines';
+
+// This is copied from core.scale.js, which is also required here and im not sure where it should be placed for both of them to access it
+function getLineValue(scale, index, offsetGridLines) {
```

Comment:
I think this is now called `getPixelForGridLine`. The code no longer quite matches. I'd recommend making it an attribute of a class and marking it private by prefixing it with `_` and adding a `@private` jsdoc tag so that it can be reused rather than copied here